kernelci/config/rootfs.py

```diff
@@ -0,0 +1,124 @@
+# along with this library; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+import yaml
+
+from kernelci.config import YAMLObject
+
+
+class RootFS(YAMLObject):
+    def __init__(self, name, rootfs_type):
+        self._name = name
+        self._rootfs_type = rootfs_type
+
+    @classmethod
+    def from_yaml(cls, rootfs, kw):
+        return cls(**kw)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def rootfs_type(self):
+        return self._rootfs_type
+
+
+class RootFS_Debos(RootFS):
+    def __init__(self, name, rootfs_type, debian_release=None,
+                 arch_list=None, extra_packages=None,
+                 extra_packages_remove=None,
+                 extra_files_remove=None, script="",
+                 docker_image=None):
+
+        super(RootFS_Debos, self).__init__(name, rootfs_type)
+        self._debian_release = debian_release
+        self._arch_list = arch_list or list()
+        self._extra_packages = extra_packages or list()
+        self._extra_packages_remove = extra_packages_remove or list()
+        self._extra_files_remove = extra_files_remove or list()
+        self._script = script
+        self._docker_image = docker_image
+
+    @classmethod
+    def from_yaml(cls, config, name):
+        kw = name
+        kw.update(cls._kw_from_yaml(
+            config, ['name', 'debian_release', 'arch_list',
+                     'extra_packages', 'extra_packages_remove',
+                     'extra_files_remove', 'script', 'docker_image']))
+        return cls(**kw)
+
+    @property
+    def debian_release(self):
+        return self._debian_release
+
+    @property
+    def arch_list(self):
+        return list(self._arch_list)
+
+    @property
+    def extra_packages(self):
+        return ' '.join(list(self._extra_packages))
+
+    @property
+    def extra_packages_remove(self):
+        return ' '.join(list(self._extra_packages_remove))
+
+    @property
+    def extra_files_remove(self):
+        return ' '.join(list(self._extra_files_remove))
+
+    @property
+    def script(self):
+        return self._script
+
+    @property
+    def docker_image(self):
+        return self._docker_image
+
+
+class RootFSFactory(YAMLObject):
+    _rootfs_types = {
+        'debos': RootFS_Debos
+    }
+
+    @classmethod
+    def from_yaml(cls, name, rootfs):
+        rootfs_type = rootfs.get('rootfs_type')
+        kw = {
+            'name': name,
+            'rootfs_type': rootfs_type,
+        }
+        rootfs_cls = cls._rootfs_types[rootfs_type] if rootfs_type else RootFS
+        return rootfs_cls.from_yaml(rootfs, kw)
+
+
+def from_yaml(yaml_path):
+    with open(yaml_path) as f:
+        data = yaml.safe_load(f)
+
+    rootfs_configs = {
+        name: RootFSFactory.from_yaml(name, rootfs)
+        for name, rootfs in data['rootfs_configs'].iteritems()
```

Comment:
`.iteritems()` is a Python 2 thing, we're using Python 3 now.

Reply:
Obiously, for some reason I didn't take the Python3 migration into consideration. changed to `.items()`